Subject: [PATCH] i386: Fix ___tls_get_addr build failure without HAVE_GCC_IFUNC

When HAVE_GCC_IFUNC is not defined (e.g., on i586 targets where GCC
does not support __attribute__((ifunc(...)))), the build fails with:

  error: '__EI____tls_get_addr' aliased to undefined symbol '__GI____tls_get_addr'

This happens because:
1. libc_ifunc_redirected() creates the IFUNC using assembly directives
   (.type name, @gnu_indirect_function) with symbol name ___tls_get_addr
2. rtld_hidden_def() expects __GI____tls_get_addr to exist, but it doesn't
   because the IFUNC was created with ___tls_get_addr, not __GI____tls_get_addr

The fix is to only use rtld_hidden_proto/rtld_hidden_def when
HAVE_GCC_IFUNC is defined, since only then does the ifunc mechanism
properly integrate with the hidden symbol aliasing.

Signed-off-by: Build Fix <build@local>
---
 sysdeps/i386/dl-tls-get-addr.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/sysdeps/i386/dl-tls-get-addr.c
+++ b/sysdeps/i386/dl-tls-get-addr.c
@@ -57,8 +57,12 @@
    an additional underscore at the beginning.  The Sun version uses
    the normal calling convention.  */

+#ifdef HAVE_GCC_IFUNC
+/* Only use hidden aliases when GCC ifunc attribute is available,
+   otherwise the symbol aliasing is incompatible with asm-based ifunc.  */
 rtld_hidden_proto (___tls_get_addr)
 rtld_hidden_def (___tls_get_addr)
+#endif

 void *
 __tls_get_addr (tls_index *ti)
